android/src/main/kotlin/com/gdelataillade/alarm/alarm/AlarmService.kt

```diff
@@ -0,0 +1,313 @@
+        intent: Intent?,
+        flags: Int,
+        startId: Int,
+    ): Int {
+        if (!channelInitialized) {
+            initChannel()
+        }
+
+        val id = intent?.getIntExtra("id", -1) ?: -1
+        if (id == -1) {
+            Log.d("flutter/AlarmService", "No id provided")
+            return START_NOT_STICKY
+        }
+
+        val action = intent?.action
+        when (action) {
+            START_ALARM_ACTION -> {
+                Log.d("flutter/AlarmService", "START alarm $id")
+                startAlarm(id, intent.getExtras())
+                return START_STICKY
+            }
+
+            STOP_ALARM_ACTION -> {
+                Log.d("flutter/AlarmService", "STOP alarm $id")
+                stopAlarm(id)
+                val recurring = intent?.getBooleanExtra("recurring", false) ?: false
+                if (recurring && intent.getExtras() != null) {
+                    scheduleRecurringAlarm(id, intent.getExtras()!!)
+                }
+                notifyAlarmDismissed(id)
+                return START_NOT_STICKY
+            }
+
+            SNOOZE_ALARM_ACTION -> {
+                Log.d("flutter/AlarmService", "SNOOZE alarm $id")
+                stopAlarm(id)
+
+                if (intent?.getExtras() == null) {
+                    Log.d("flutter/AlarmService", "Cannot snooze alarm - no extras provided")
+                    return START_NOT_STICKY
+                }
+
+                val snoozeDuration = intent?.getIntExtra("snoozeDuration", 5 * 60) ?: 5 * 60 // default to 5 minutes
+                val rescheduleTime = System.currentTimeMillis() + snoozeDuration * 1000
+                intent.putExtra("dateTime", rescheduleTime)
+                val extras = intent.getExtras()!!
+                alarmHandler.scheduleAlarm(rescheduleTime, id, extras)
+                notifyAlarmSnoozed(id)
+
+                // Save alarm
+                storageHandler.saveAlarm(id, extras)
+                return START_NOT_STICKY
+            }
+
+            else -> {
+                Log.d("flutter/AlarmService", "Unknown action: $action")
+                return START_NOT_STICKY
+            }
+        }
+    }
+
+    private fun startAlarm(
+        id: Int,
+        extras: Bundle?,
+    ) {
+        showNotification(id, extras)
+
+        // TODO(system-alert): Decide if this is something we want to do
+        // val activityIntent = applicationContext.packageManager.getLaunchIntentForPackage(applicationContext.packageName)
+        // activityIntent?.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
+        // startActivity(activityIntent)
+
+        startAlarmAudio(id, extras)
+        notifyAlarmRinging(id)
+
+        // Wake up the device
+        val wakeLock =
+            (getSystemService(Context.POWER_SERVICE) as PowerManager)
+                .newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, "app:AlarmWakelockTag")
+        wakeLock.acquire(5 * 60 * 1000L) // 5 minutes
```

Comment:
That might be a good idea. I just copied it from here: https://github.com/gdelataillade/alarm/blob/main/android/src/main/kotlin/com/gdelataillade/alarm/alarm/AlarmService.kt#L107 assuming they knew what they were doing...

I'll move it up to prevent going to sleep while setting the alarm and use `setWakeMode` on the `mediaPlayer` to let it manage that for us.